src/components/Details/Legislation/LegislativeHistoryNode.tsx

```diff
@@ -0,0 +1,155 @@
+  files?: Document[];
+}
+
+const LegislativeHistoryNode: FC<LegislativeHistoryNodeProps> = ({
+  eventMinutesItem,
+  isLastIndex,
+}: LegislativeHistoryNodeProps) => {
+  const isMobile = useMediaQuery({ query: `(max-width: ${screenWidths.tablet})` });
+  const MARGIN = isMobile ? 4 : 12;
+
+  const ReferenceFrame = styled.div({ position: "relative" });
+  const ExpandingContainer = styled.div({
+    margin: MARGIN,
+    display: "flex",
+    flexDirection: "row",
+    gap: MARGIN * 3,
+  });
+  const TextContainer = styled.div({ flex: 1 });
+  const SummaryInfo = styled.div({ display: "flex", flexDirection: "row" });
+  const VotingGraphicContainer = styled.div<{ marginLeft: number }>((props) => ({
+    flex: 1,
+    marginTop: MARGIN,
+    marginLeft: props.marginLeft,
+  }));
+
+  const ConnectBar = styled.div({
+    position: "absolute",
+    top: MARGIN + DOT_SIZE,
+    left: DOT_SIZE / 2 + MARGIN,
+    height: "100%",
+    width: 2,
+    zIndex: 0,
+    backgroundColor: "grey",
+  });
+
+  const { firebaseConfig } = useAppConfigContext();
+  const { language } = useLanguageConfigContext();
+  const localizedDateString = eventMinutesItem.event?.event_datetime.toLocaleDateString(language, {
+    month: "long",
+    day: "numeric",
+    year: "numeric",
+  });
+
+  const fetchDataForNode = useCallback(async () => {
+    const data: LegislativeHistoryNodeData = {
+      votes: [],
+      files: [],
+    };
+    const eventMinutesItemFileService = new EventMinutesItemFileService(firebaseConfig);
+    const minutesItemFiles =
+      await eventMinutesItemFileService.getEventMinutesItemFilesByEventMinutesItemId(
+        eventMinutesItem.id
+      );
+    data.files = minutesItemFiles.map(({ name, uri }) => {
+      return {
+        label: name,
+        url: uri,
+      };
+    });
+    if (eventMinutesItem.decision) {
+      const votesService = new VoteService(firebaseConfig);
+      data.votes = await votesService.getVotesByEventMinutesItemId(eventMinutesItem.id);
+    }
+
+    return Promise.resolve(data);
+  }, [eventMinutesItem.id, eventMinutesItem.decision, firebaseConfig]);
+
+  const { state: votesDataState } = useFetchData<LegislativeHistoryNodeData>(
+    { ...initialFetchDataState },
+    fetchDataForNode
+  );
+
+  return (
+    <ReferenceFrame>
+      <ExpandingContainer>
+        <Dot className={"cdp-bg-neutral-grey"} />
+        <TextContainer>
+          <SummaryInfo>
+            <Link to={`/events/${eventMinutesItem.event?.id}`}>{localizedDateString}</Link>
+            {eventMinutesItem.decision && (
+              <DecisionResult
+                result={eventMinutesItem.decision}
+                style={{
+                  marginLeft: MARGIN,
+                  display: "flex",
+                  flex: 1,
+                  alignItems: "center",
+                }}
+              />
+            )}
+          </SummaryInfo>
+          <LazyFetchDataContainer
+            data="legislativeHistoryNodeDataState"
```

Comment:
the `data` prop is used to display message like
- `Fetching <data>...`
- `No <data> found.`

Could you change the text to something more appropriate?

Reply:
Yeah. I added a comment to the data prop, for clarity.